src/main/java/com/algorand/algosdk/util/abi/Type.java

```diff
@@ -130,6 +131,21 @@ else if (currentIndex != 0 && delta < 0)
         return until;
     }
 
+    static Object[] unifyToArrayOfObjects(Object val) {
+        if (val.getClass().isArray()) {
+            if (val instanceof Object[])
+                return (Object[]) val;
+            int length = Array.getLength(val);
+            Object[] outputArray = new Object[length];
+            for (int i = 0; i < length; ++i)
+                outputArray[i] = Array.get(val, i);
+            return outputArray;
+        } else if (val instanceof List<?>)
+            return ((List<?>) val).toArray(new Object[0]);
+        else
+            throw new IllegalArgumentException("cannot infer type for unify array/list-like object to object array");
+    }
+
     public static byte[] getLengthEncoded(byte[] encoded) {
```

Comment:
Can you make `getLengthEncoded`, `getContentEncoded`, and `castToTupleType` package local as well?

Reply:
Once you add this, it looks good to merge